src/GraphMetadataViewExtension/GraphMetadataViewExtension.cs

```diff
@@ -53,21 +54,33 @@ private void MenuItemCheckHandler(object sender, RoutedEventArgs e)
         #region Storage Access implementation
         public void OnWorkspaceOpen(Dictionary<string, string> extensionData)
         {
-            throw new NotImplementedException();
+            foreach (var p in this.viewModel.CustomProperties)
```

Comment:
In workspace open we dont want to add to `extensionData` instead we should check if there are any custom properties saved in dictionary and if there is we should create new `CustomProperty` and add it to the collection